specviz/plugins/model_editor/items.py

```diff
@@ -24,38 +16,15 @@ def flux(self):
 
         result = self.model_editor_model.evaluate()
 
-        flux_units = self.data(self.DataRole).flux.unit
-
-        if self._plot_data_item is None:
-            model_flux_units = flux_units
-        else:
-            model_flux_units = u.Unit(self._plot_data_item.data_unit)
-
         if result is not None:
-            model_flux = result(self.model_spectral_axis.value) * model_flux_units
-            return model_flux.to(flux_units,
-                                 equivalencies=u.equivalencies.spectral_density(
-                                     self.model_spectral_axis))
-
-        return np.zeros(self.spectral_axis.size) * model_flux_units
+            return result(self.spectral_axis.value) * self.data(self.DataRole).flux.unit
```

Comment:
If the user enters the model parameters assuming the same units as displayed by the plot widget, the values returned by `self.model_editor_model.evaluate()` will not match that. I agree getting the units from the plot item is not a good solution, but we need to have a way of keeping track of what units were used to fit the model until SpecUtils modeling is updated.

Reply:
The units are on the spectrum passed to the fitter are now set in the model editor plugin. Thus, the units returned should also be in the plot units.